src/deque.py

```diff
@@ -0,0 +1,71 @@
+"""Implementation of Deque."""
+
+from dll import DoublyLinkedList
+
+
+class Deque(object):
+    """Class implementation of queue.
+
+    append(val): adds value to the end of the deque.
+    appendleft(val): adds a value to the front of the deque.
+    pop(): removes a value from the end of the deque and returns it
+            (raises an exception if the deque is empty)
+    popleft(): removes a value from the front of the deque and returns it
+            (raises an exception if the deque is empty)
+    peek(): returns the next value that would be returned by pop but
+            leaves the value in the deque (returns None if the deque is empty)
+    peekleft(): returns the next value that would be returned by popleft
+            but leaves the value in the deque (returns None if the deque
+            is empty)
+    size(): returns the count of items in the queue (returns 0 if the
+            queue is empty)
+
+    """
+
+    def __init__(self, iterable=None):
+        """Instatiate Queue."""
+        self.dll = DoublyLinkedList(iterable)
+        self.head_node = self.dll.head_node
+        self.tail_node = self.dll.tail_node
+        self.length = self.dll.length
+
+    def append(self, contents):
+        """Add value to the end of the deque."""
+        self.dll.append(contents)
+        self.head_node = self.dll.head_node
+        self.tail_node = self.dll.tail_node
+        self.length = self.dll.length
+
+    def appendleft(self, contents):
+        """Add value to the front of the deque."""
+        self.dll.push(contents)
+        self.head_node = self.dll.head_node
+        self.tail_node = self.dll.tail_node
+        self.length = self.dll.length
+
+    def pop(self):
+        """Remove and return the current tail node."""
+        return self.dll.shift()
+        self.head_node = self.dll.head_node
+        self.tail_node = self.dll.tail_node
+        self.length = self.dll.length
```

Comment:
I suspect the deque pop / dll shift method might be not be working as expected...
```python
In [2]: dq = Deque([1, 2, 3, 4])

In [3]: dq.pop()
Out[3]: 1

In [4]: dq.pop()
---------------------------------------------------------------------------
ValueError: Linked list is already empty
```

while this works:
```python
In [2]: dq = Deque([1, 2, 3, 4])

In [3]: dq.popleft()
Out[3]: 4

In [4]: dq.popleft()
Out[4]: 3

In [5]: dq.popleft()
Out[5]: 2

In [6]: dq.popleft()
Out[6]: 1
````